src/libsystemd/sd-netlink/netlink-types-genl.c

```diff
@@ -221,15 +221,26 @@ static const NLType genl_wireguard_types[] = {
         [WGDEVICE_A_PEERS]       = { .type = NETLINK_TYPE_NESTED, .type_system = &genl_wireguard_peer_type_system },
 };
 
+/***************** genl NetLabel type systems *****************/
+static const NLType genl_netlabel_types[] = {
+        [NLBL_UNLABEL_A_IPV4ADDR] = { .type = NETLINK_TYPE_IN_ADDR },
+        [NLBL_UNLABEL_A_IPV4MASK] = { .type = NETLINK_TYPE_IN_ADDR },
```

Comment:
Not sure, but maybe, `.size = sizeof(struct in_addr)`??

Reply:
OK. Grepping for other uses of `NETLINK_TYPE_IN_ADDR` shows that the `.size` isn't used very consistently, even for IPv6:
src/libsystemd/sd-netlink/netlink-types-genl.c:119:        [FOU_ATTR_LOCAL_V4]          = { .type = NETLINK_TYPE_IN_ADDR },
src/libsystemd/sd-netlink/netlink-types-genl.c:121:        [FOU_ATTR_LOCAL_V6]          = { .type = NETLINK_TYPE_IN_ADDR },
src/libsystemd/sd-netlink/netlink-types-genl.c:149:        [L2TP_ATTR_IP_DADDR]          = { .type = NETLINK_TYPE_IN_ADDR },
src/libsystemd/sd-netlink/netlink-types-genl.c:153:        [L2TP_ATTR_IP6_DADDR]         = { .type = NETLINK_TYPE_IN_ADDR },

Maybe they should be fixed too?

Reply:
Yeah, please if you'd like.

Reply:
Made PR #23302.